### ISSUE TITLE
gtag() compatibility

### ISSUE THREAD

**Issue by androidacy-user** [jahilldev/minimal-analytics#30]
Hello!

We'd like to start using this library for GA4 on our site(s), but we're hitting a big snag right now: having to restructure all of our tracking calls to use mostly custom code.

Is there a plan to add gtag() or even dataLayer compatibility in the future?

**Comment by jahilldev** [comment]
Hi @androidacy-user!

Thanks for reaching out 👍 

I would suggest this could be achieved by using a custom abstraction in your application. From memory, `gtag` is just a global function that wraps various GA triggers and config. If your application uses this directly, without your own abstraction (I would _strongly_ suggest not doing this in future, it makes changing third party integrations difficult, as you're find out!), then you could simple write your own `gtag` function that wraps this package's `track()` function.

Something like this:

```js
import { track } from '@minimal-analytics/ga4';

/*[...]*/
const GA_ID = 'G-XXXXXXXX';

window.gtag = (method, ...args) => {
  if(method === 'config' && args[0] === GA_ID) {
    track(GA_ID) // page_view
  }

  if(method === 'event') {
    track({ type: args[0], event: args[1] }); // custom event

    return;
  }
  
  /*[...]*/ 
 // etc, etc :)
};
```

This would be a 1-1 proxy for triggering an event using `gtag`, e.g:

***Page View***
```js
// maybe hold the measurement ID in a global var
gtag('config', 'G-XXXXXXXX');
```

***Custom Event***
```js
gtag('event', 'login', {'ep.user': 'John Smith'});
```

You'll need to cater to all uses of `gtag` throughout your application, but with a little effort this shouldn't be too tricky.

Ideally your application would use it's own analytics abstraction, so if you wanted to switch from say GA to something completely different, you'd just refactor the abstraction, not the usage.  This of course, can be said for any third party integration.

We won't be adding `gtag` compatibility to this package, given the multitiude of ways Google Analytics can be integrated, it would dramatically (and gradually) increase the overall size of the code, whilst also potentially opening us up to supporting deprecated methods / API's in future. This is really a consumer issue, and should likely be handled by a given application.

I hope this helps! Please let me know if you have any further questions, or require help with anything.

It's important to note, if your application is **not** using a Node environment, we currently don't expost the `track` function globally. If this is the case, let me know, and we can look into doing that for you.

All the best! 👍

**Comment by jahilldev** [comment]
@androidacy-user In that case, if you're not using a build tool for your frontend, then for the above to work we'll need to expose the `track` function globally for you.

Once we've done that, something like `window.track()` or similar, you can follow the above suggested abstraction, minus the import statement.

I'll see if I can get a release done sometime this week for you. This has been on my watch list, but nobody thus far has piped up to say they want it, but now you have!

Thanks again for contributing 👍

**Comment by jahilldev** [comment]
@androidacy-user Ok I've opened a PR that introduces an additional config property, `defineGlobal`.
https://github.com/jahilldev/minimal-analytics/pull/31

Once this is merged / deployed you'll have access to the `track()` function on the Window.

**Comment by jahilldev** [comment]
@androidacy-user I've just deployed the latest version that provides the global function definition, [v1.8.4](https://github.com/jahilldev/minimal-analytics/releases/tag/ga4%401.8.4).

Let me know how you get on!

**Comment by jahilldev** [comment]
@androidacy-user Apologies, I've just identified a bug. Fixed now in [v1.8.5](https://github.com/jahilldev/minimal-analytics/releases/tag/ga4%401.8.5)!
